src/google-maps/map-polyline/map-polyline.ts

```diff
@@ -0,0 +1,234 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// Workaround for: https://github.com/bazelbuild/rules_nodejs/issues/1265
+/// <reference types="googlemaps" />
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewEncapsulation
+} from '@angular/core';
+import {BehaviorSubject, combineLatest, Observable, Subject} from 'rxjs';
+import {map, take, takeUntil} from 'rxjs/operators';
+
+import {GoogleMap} from '../google-map/google-map';
+
+/**
+ * Angular component that renders a Google Maps Polyline via the Google Maps JavaScript API.
+ * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline
+ */
+@Component({
+  moduleId: module.id,
+  selector: 'map-polyline',
+  template: '<ng-content></ng-content>',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class MapPolyline implements OnInit, OnDestroy {
+  @Input()
+  set options(options: google.maps.PolylineOptions) {
+    this._options.next(options || {});
+  }
+
+  @Input()
+  set path(path: google.maps.MVCArray<google.maps.LatLng>|google.maps.LatLng[]|
+           google.maps.LatLngLiteral[]) {
+    this._path.next(path);
+  }
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.click
+   */
+  @Output() mapClick = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.dblclick
+   */
+  @Output() mapDblclick = new EventEmitter<google.maps.PolyMouseEvent>();
```

Comment:
Why are all of these events prefixed with `map`? It makes it sound like it'll apply to the entire map whereas, according to the linked docs, it's only events in the polygon.

Reply:
Originally, with the first Google Map components, I did not have the map prefix, but there were unforeseen difficulties with overwriting the default mouse event handlers. It became necessary to separate their behaviors into "click" and "mapClick" classes. We could have used the prefix of whatever component was being used, "markerClick" or "polylineClick" but I thought using "mapClick" for everything indicated that the behavior was different from the regular "click" behavior but consistent among all Google Map components.

Reply:
We discussed this in Slack and since `polylineClick` is of type `PolyMouseEvent `and `markerClick` is of type `MouseEvent`, we decided that it would be best to prefix each of these click events with the component name.

Reply:
That makes sense. The behavior is a bit different for clicks for polyline, so I've gone ahead and changed it. Do you think we should use a different prefix for every component or just ones with different behavior like the PolyMouseEvent?

Reply:
I think we should use a different prefix for each one with a different behavior at minimum. 

I don't see a problem with using the different component prefix with each one,  but if there is a concern, then I don't think it's a big deal either way.